lib/rules/no-boolean-default.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * @fileoverview Prevents boolean defaults from being set
+ * @author Hiroki Osame
+ */
+'use strict'
+
+const utils = require('../utils')
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'disallow boolean defaults',
+      category: undefined,
+      recommended: false,
+      url: 'https://github.com/vuejs/eslint-plugin-vue/blob/v5.0.0-beta.3/docs/rules/no-boolean-default.md'
+    },
+    fixable: 'code',
+    schema: [
+      {
+        enum: ['default-false', 'no-default', 'constructor']
+      }
+    ]
+  },
+
+  create (context) {
+    function getPropsDef (vueComp) {
+      return vueComp.properties.find(p =>
+        p.type === 'Property' &&
+            p.key.type === 'Identifier' &&
+            p.key.name === 'props' &&
+            p.value.type === 'ObjectExpression'
+      )
+    }
+
+    function getBooleanProps (propsDef) {
+      return propsDef.value.properties
+        .filter(p => {
+          return (p.type === 'Property') && (p.value.type === 'ObjectExpression') && p.value.properties.find(p =>
+            (
+              p.type === 'Property' &&
+              p.key.type === 'Identifier' &&
+              p.key.name === 'type' &&
+              p.value.type === 'Identifier' &&
+              p.value.name === 'Boolean'
+            )
+          )
+        })
+    }
+
+    function getDefaultNode (propDef) {
+      return propDef.value.properties.find(p => {
+        return (p.type === 'Property' && p.key.type === 'Identifier' && p.key.name === 'default')
+      })
+    }
+
+    return utils.executeOnVueComponent(context, (obj) => {
+      const propsDef = getPropsDef(obj)
```

Comment:
If you decide to merge from master, you can simplify this code

```suggestion
      const propsDef = utils.getComponentProps(obj)
        .filter(prop => prop.value && prop.value.type === 'ObjectExpression')
```